components/connectivity-certs-controller/internal/certificates/csrprovider_test.go

```diff
@@ -0,0 +1,187 @@
+xKySs3fWcl167+ZKrBzXkRj9LGs6yuP9AHIP5gGj3a60/ZoGsUfiEsj/pBN7y9KN
+WwqZHOPiuHzg0OLXI0Y6A3Of8y7+WrEAyS0ck7Ue5TTxRKKmJJfF7DE7GkTw8I0H
+dMad3pJe/aPuDIPRiWxCVDiL9QGcejstBvTz4wIDAQABAoIBAD9lXaOxIHEjuLlO
+UGNfm/OMIXZfvY5hb/cClDxttCzhJQKG7HK/fwwaMc6laohKvV8B9ScTxTx3rUS7
+2AxAwIVKU8xMxqaCILnkS5ylwhxJXzBJdKKZBRyxOt/C+8+V0NrWk2Z3YyaKtUMR
+9hisqhEKXoXv/FYojPV4qJL+QZNvQ+9kI7kBinred5tz5xKk2BHzYs+Em37A01NP
+Mn9oftth4H1OtGQkE2mBKjt8vjP2IKVnoG/yGKbTU6C/wkXNnB0cABSdyNquJlbp
+T9CVpx2qP173h0yzEAPHSkQe9DPP6JHQSgJZpGUEdoiQe5vKxmGqvRfPbq1UTo/j
+jHs/etkCgYEA8zaDNk+vVAXYTjexJy5yT9cIFN8FrLloD0PkLGpEJlBMGhTLDATS
+XlJrt6EFqq2abBPaiLkynF2+qvHr2WQlFA0NJHaYlcTZHE10WtFlwwlM9gEm3YeK
+jYLwQmfigaB3Ut4ckr3NnZa26LSODb1rpSuWqXZMSfDN2TiaEMtcJ4UCgYEA6E8o
+8/UFB8mOXSKfkHC3n/QUmcAhjVqgCF9L/6ht4NQouRRObe1kKQylh2JejetNvZLk
+99dnFK6OErevHAv1j0PcBilTq2dXmKFVhkUp9bFj3Kp3ALzinZAwbV9GTWiA+3Wv
+rgaLdxJxTR8Cly8JWR1YuKrWj5oCIMmFoqk7ZkcCgYEAkyTt6ZP4PVtz7I6hLVVa
+b5dnGkl8A24A2Qt4Jq78IDoAcN8XoWPhapNu/B/9b6+sd6rjUkjJp/THgGDxEgsW
+q7ThuKfP1PzNZeQueyuo54DfAQ7dVrXES61mcqarUUWmK4qZuuX+WlNuwgdK0mFB
+mSJv+oLJ0QpRYBRwkayXSokCgYAPq89ibaPyO4mMBNrovoHUm32MRaa9x2BGUE9r
+JqyK3yUEHzePONVp432DHYKtZjMvV6p0gaZlgcT5xEReyvu8t2IvVDhdtrH1DOUd
+Eqta9KV87E7s0NEkueZaanPuot8Yl37LaYuc87SK9E2Tb0vdJBqpEnU46LW+Cnom
+V+423wKBgBYFjwG387TT/DudzxbSvrWLGfEddbZM5sH4YcvmqBqu/qSssqd4ueOs
+tOetczdVd2on5cQJ9WTvye2s066GRSIcPEnh+DQZTea350HkzHSb3+naylQ84BL3
+AZwS8JvJu+rnTnjRcfcvrPYjIigwMzgB1vun3zJz1h4J9s9uCWjZ
+-----END RSA PRIVATE KEY-----`
+
+		secretData := map[string][]byte{
+			clusterKeySecretKey: []byte(privateKey),
+		}
+
+		secretRepository := &mocks.Repository{}
+		secretRepository.On("Get", clusterSecretName).Return(secretData, nil)
+
+		csrProvider := NewCSRProvider(clusterSecretName, "", secretRepository)
+
+		// when
+		csr, err := csrProvider.CreateCSR(plainSubject)
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, csr)
+
+		receivedCSR := decodeCSR(t, csr)
+
+		require.NotNil(t, receivedCSR)
+		assertSubject(t, receivedCSR)
+		secretRepository.AssertExpectations(t)
+	})
+
+	t.Run("should create CSR when no secret", func(t *testing.T) {
+		// given
+		secretRepository := &mocks.Repository{}
+		secretRepository.On("Get", clusterSecretName).Return(nil, k8serrors.NewNotFound(schema.GroupResource{}, "error"))
+		secretRepository.On("Override", clusterSecretName, mock.AnythingOfType("map[string][]uint8")).Return(nil).
+			Run(func(args mock.Arguments) {
+				secretData, ok := args[1].(map[string][]byte)
+				require.True(t, ok)
+				assert.NotEmpty(t, secretData[clusterKeySecretKey])
+			})
+
+		csrProvider := NewCSRProvider(clusterSecretName, "", secretRepository)
+
+		// when
+		csr, err := csrProvider.CreateCSR(plainSubject)
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, csr)
+
+		receivedCSR := decodeCSR(t, csr)
+
+		require.NotNil(t, receivedCSR)
+		assertSubject(t, receivedCSR)
+		secretRepository.AssertExpectations(t)
+	})
+
+	t.Run("should create CSR when no key in secret", func(t *testing.T) {
```

Comment:
We should also cover the case when secret exists but private key under clusterKeySecretKey is corrupted.
```
block, _ := pem.Decode(secret[clusterKeySecretKey])
if block == nil {
	return cp.createClusterKeySecret()
} 
```

Reply:
Technically it is the same case as  `pem.Decode` should also return `nil`, but I have added an additional test.